### ISSUE TITLE
Model Eval

### ISSUE THREAD

**Issue by Deepblue129** [IBM/pytorch-seq2seq#33]
Evaluator does not use the evaluation mode for the model

`http://pytorch.org/docs/master/nn.html#torch.nn.Module.eval`

Without eval mode, dropout is enabled during inference.

**Comment by kylegao91** [comment]
`module.eval()` is equivalent to `module.train(False)`.  It is correctly used in the trainer for evaluating develop set.  I think it's a good idea to add it to the evaluator as the first thing it checks, and so that the trainer doesn't have to set it.
